demisto_sdk/commands/common/docker_helper.py

```diff
@@ -235,13 +239,22 @@ def create_container(
         """
         Creates a container and pushing requested files to the container.
         """
-        container: docker.models.containers.Container = (
-            init_global_docker_client().containers.create(
-                image=image, command=command, environment=environment, **kwargs
+        docker_client = init_global_docker_client()
+
+        try:
+            container: docker.models.containers.Container = (
+                docker_client.containers.create(
+                    image=image, command=command, environment=environment, **kwargs
+                )
             )
-        )
+        except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as e:
+            if container_name := kwargs.get("container_name"):
+                docker_client.containers.get(container_name).remove(force=True)
```

Comment:
in case we didn't create the container with a specific name, we should try to remove the container that was created with the image, meaning we should get the container model by the image and not only by the name.